pkg/ingester/ingester.go

```diff
@@ -195,6 +196,13 @@ func New(cfg Config, clientConfig client.Config, limits *validation.Overrides, c
 		return nil, err
 	}
 
+	// Init the limter and instantiate the user states which depend on it
+	i.limiter = NewSeriesLimiter(limits, i.lifecycler, cfg.LifecyclerConfig.RingConfig.ReplicationFactor, cfg.ShardByAllLabels)
+	i.userStates = newUserStates(i.limiter, cfg)
+
+	// Now that user states have been created, we can start the lifecycler
+	i.lifecycler.Start()
```

Comment:
Curious why this needs to be done after the user states have been created. Why can't we just start when we create the lifecycler as before?

Reply:
The ingester depends on the lifecycler, but the `Lifecycler` depends on the ingester instance too for the flush + transfer. Due to this cyclic dependency, we currently create the ingester instance first and pass it to the lifecycler.

Once the lifecycler is created, it also start the `loop()`, which may potentially lead to some ingester's code requiring `i.userStates` which is not guaranteed to be instantiated yet (after changes in this PR).

**Right now, this should never happen**, but in the future we may do code changes leading to it, which may be difficult to spot in code reviews.

The change proposed is to be more on the safe side, but I'm OK rolling it back if you don't see the value.